azurerm/internal/services/apimanagement/api_management_subscription_resource.go

```diff
@@ -227,8 +228,9 @@ func resourceApiManagementSubscriptionRead(d *pluginsdk.ResourceData, meta inter
 		d.Set("state", string(props.State))
 		productId := ""
 		apiId := ""
-		if *props.Scope != "" {
-			// the scope is either a product or api id or "all_apis" constant
+		// check if the subscription is for all apis or a specific product/ api
+		if *props.Scope != "" && !strings.HasSuffix(*props.Scope, "/apis") {
```

Comment:
looks like there's been a crash here actually:

```suggestion
		if props.Scope != nil && *props.Scope != "" && !strings.HasSuffix(*props.Scope, "/apis") {
```